ouroboros-network-protocols/src/Ouroboros/Network/Protocol/CBOR.hs

```diff
@@ -0,0 +1,18 @@
+{-# LANGUAGE RankNTypes #-}
+
+module Ouroboros.Network.Protocol.CBOR where
+
+import qualified Codec.CBOR.Decoding as CBOR
+import qualified Codec.CBOR.Encoding as CBOR
+import qualified Codec.Serialise as CBOR
+
+data CBORCodec a b = CBORCodec {
+    encodeCBOR :: a -> CBOR.Encoding,
+    decodeCBOR :: forall s. CBOR.Decoder s b
+  }
```

Comment:
I decided to go for different types because in some parts of the code we require an encoder of a type and a decoder of other type and in this way I could group that particular case in this type

Reply:
[`c2298c1` (#4430)](https://github.com/input-output-hk/ouroboros-network/pull/4430/commits/c2298c14e5e5e7fccce67a3b13d80b89d19de72c#diff-150eb11e1d1f827c453ca5290a44cb95ddf982b5641950e412612184bafe8bd0R40) here for example.

Reply:
https://github.com/input-output-hk/ouroboros-network/pull/4430/files#diff-150eb11e1d1f827c453ca5290a44cb95ddf982b5641950e412612184bafe8bd0R40

Reply:
I like it ! :)

Reply:
Done!